sfputil/main.py

```diff
@@ -1991,11 +1992,62 @@ def loopback(port_name, loopback_mode):
         click.echo("{}: This functionality is not implemented".format(port_name))
         sys.exit(ERROR_NOT_IMPLEMENTED)
 
+    namespace = multi_asic.get_namespace_for_port(port_name)
+    config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+    if config_db is not None:
+        config_db.connect()
+        try:
+            subport = int(config_db.get(config_db.CONFIG_DB, f'PORT|{port_name}', 'subport'))
+        except TypeError:
+            click.echo(f"{port_name}: subport is not present in CONFIG_DB")
+            sys.exit(EXIT_FAIL)
+
+        # If subport is set to 0, assign a default value of 1 to ensure valid subport configuration
+        if subport == 0:
+            subport = 1
+    else:
+        click.echo(f"{port_name}: Failed to connect to CONFIG_DB")
+        sys.exit(EXIT_FAIL)
+
+    state_db = SonicV2Connector(use_unix_socket_path=False, namespace=namespace)
+    if state_db is not None:
+        state_db.connect(state_db.STATE_DB)
+        try:
+            host_lane_count = int(state_db.get(state_db.STATE_DB,
+                                               f'TRANSCEIVER_INFO|{port_name}',
+                                               'host_lane_count'))
+        except TypeError:
+            click.echo(f"{port_name}: host_lane_count is not present in STATE_DB")
+            sys.exit(EXIT_FAIL)
+
+        try:
+            media_lane_count = int(state_db.get(state_db.STATE_DB,
+                                                f'TRANSCEIVER_INFO|{port_name}',
+                                                'media_lane_count'))
+        except TypeError:
+            click.echo(f"{port_name}: media_lane_count is not present in STATE_DB")
+            sys.exit(EXIT_FAIL)
+    else:
+        click.echo(f"{port_name}: Failed to connect to STATE_DB")
+        sys.exit(EXIT_FAIL)
+
+    if 'host-side' in loopback_mode:
+        lane_mask = ((1 << host_lane_count) - 1) << ((subport - 1) * host_lane_count)
+    elif 'media-side' in loopback_mode:
+        lane_mask = ((1 << media_lane_count) - 1) << ((subport - 1) * media_lane_count)
+    else:
+        lane_mask = 0
```

Comment:
@xinyulin can we write a generic function `get_subport_lane_mask(subport, lane_count)`  which will return the lane mask for the particular `subport`

Reply:
@prgeor Yes, it has been added in the latest commit.